src/core/settings/qgssettingsentry.cpp

```diff
@@ -55,37 +55,89 @@ QString QgsSettingsEntryBase::key( const QStringList &dynamicKeyPartList ) const
   QString completeKey = mKey;
   if ( !mPluginName.isEmpty() )
   {
-    if ( !completeKey.startsWith( "/" ) )
-      completeKey.prepend( "/" );
+    if ( !completeKey.startsWith( '/' ) )
+      completeKey.prepend( '/' );
     completeKey.prepend( mPluginName );
   }
 
   if ( dynamicKeyPartList.isEmpty() )
   {
     if ( hasDynamicKey() )
-      QgsLogger::warning( QStringLiteral( "Settings '%1' have a dynamic key but the dynamic key part was not provided" ).arg( completeKey ) );
+      QgsDebugMsg( QStringLiteral( "Settings '%1' have a dynamic key but the dynamic key part was not provided" ).arg( completeKey ) );
 
     return completeKey;
   }
   else
   {
     if ( !hasDynamicKey() )
     {
-      QgsLogger::warning( QStringLiteral( "Settings '%1' don't have a dynamic key, the provided dynamic key part will be ignored" ).arg( completeKey ) );
+      QgsDebugMsg( QStringLiteral( "Settings '%1' don't have a dynamic key, the provided dynamic key part will be ignored" ).arg( completeKey ) );
       return completeKey;
     }
 
     for ( int i = 0; i < dynamicKeyPartList.size(); i++ )
     {
-      completeKey.replace( QString( "%%1" ).arg( QString::number( i + 1 ) ), dynamicKeyPartList.at( i ) );
+      completeKey.replace( QStringLiteral( "%" ).append( QString::number( i + 1 ) ), dynamicKeyPartList.at( i ) );
     }
   }
   return completeKey;
 }
 
+bool QgsSettingsEntryBase::keyIsValid( const QString &key ) const
+{
+  // Key to check
+  QString completeKeyToCheck = key;
+
+  QString settingsPrefix = QgsSettings().prefixedKey( QString(), section() );
+  settingsPrefix.chop( 1 );
+  if ( !completeKeyToCheck.startsWith( settingsPrefix ) )
+  {
+    if ( !mPluginName.isEmpty()
+         && !completeKeyToCheck.startsWith( mPluginName ) )
+    {
+      if ( !completeKeyToCheck.startsWith( '/' ) )
+        completeKeyToCheck.prepend( '/' );
+      completeKeyToCheck.prepend( mPluginName );
+    }
+
+    if ( !completeKeyToCheck.startsWith( '/' ) )
+      completeKeyToCheck.prepend( '/' );
+    completeKeyToCheck.prepend( settingsPrefix );
+  }
+
+  // Prefixed settings key
+  QString prefixedSettingsKey = definitionKey();
+  if ( !prefixedSettingsKey.startsWith( settingsPrefix ) )
+  {
+    if ( !prefixedSettingsKey.startsWith( '/' ) )
+      prefixedSettingsKey.prepend( '/' );
+    prefixedSettingsKey.prepend( settingsPrefix );
+  }
+
+  if ( !hasDynamicKey() )
+    return completeKeyToCheck == prefixedSettingsKey;
```

Comment:
We should probably use `QString::compare` method for a string comparison.

Reply:
@pblottiere Thank you for the review! Why is `QString::compare` preferable over the == operator in this case? Compare return an integer with more information (greater/smaller/equal) and all i want is to know if the string are equal.

Reply:
in this case, it doesn't seem to be significant
https://forum.qt.io/topic/52155/compare-2-qstring-use-or-qstring-compare